org.eclipse.xtext/src/org/eclipse/xtext/naming/CopyQualifiedNameService.java

```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.naming;
+
+import org.eclipse.emf.ecore.EObject;
+
+/**
+ * Replacement for org.eclipse.xtext.ui.editor.copyqualifiedname.CopyQualifiedNameService
+ * 
+ * @author Anton Kosyakov - Initial contribution and API
+ * @author Arne Deutsch - Moved to new package
+ * @since 2.14
+ */
+public interface CopyQualifiedNameService {
+
+	String getQualifiedName(EObject selectedElement, EObject context);
```

Comment:
The context parameter is not used so far in the implementation and I doubt that it makes sense at all. Please remove the parameter here and add a default method for the deprecated interface

Reply:
The context parameter allows to support use cases with aliases. Assuming you have an element with two different properties that can be names, so it has two qualified names, let's say an IP address and a URL.
When you refer to that object by ip, you probably want to copy the IP on copy QN. When you refer to it via text, you probably want to copy the URL. The context parameter is passed into the service to allow this distinction.